chirho/dynamical/handlers/interruption.py

```diff
@@ -86,15 +85,13 @@ def _pyro_apply_interruptions(self, msg) -> None:
 
 
 class _PointObservationMixin(Generic[T]):
-    data: Dict[str, Observation[T]]
+    observation: Observation[State[T]]
     time: R
 
     def _pyro_apply_interruptions(self, msg) -> None:
-        dynamics: Observable[T] = msg["args"][0]
-        current_state: State[T] = msg["args"][1]
-
-        with condition(data=self.data):
-            dynamics.observation(current_state)
+        dynamics = msg["args"][0]
+        state: State[T] = msg["args"][1]
+        msg["value"] = (dynamics, observe(state, self.observation))
```

Comment:
Would we ever expect `observe` to change its output to anything other than `state`? If it's just a side effect that changes the likelihood then maybe the following would be better?

```
state: State[T] = msg["args"][1]
observe(state, self.observation)
```

Reply:
Possibly, yes - consider a particle filter that updates particles at each observe() call